### ISSUE TITLE
Accessibility (a11y) issue from Level Access: Table element without table headers

### ISSUE THREAD

**Issue by openstaxalina** [openstax/cnx-recipes#5227]
Description
Rule description
One or more table elements that hold table data, without an ARIA-assigned role, not intentionally hidden in the DOM and available to assistive technologies, does not contain one or more th elements (without an ARIA-assigned role); th elements with a role set to: 'rowheader | columnheader'; or td elements with a role set to: 'rowheader | columnheader'.

Severity
Critical

WCAG Success criteria
[1.3.1 Info and Relationships - Level A](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships)

Why it matters
Without <code>th</code> elements (table header cells), there will be no way to associate data cells and their categories. When data cells are not properly associated with headers, users of assistive technology may not be able to reliably identify headers and navigate data tables.

What to do
Specify table headers for the table using the th element or by use of role columnheader (for column headers) or rowheader (for row headers). Upon identifying table headers, ensure that the headers are properly associated with table content.

Screenshots and attachments
[See Google Sheet with 8 affected pages here](https://docs.google.com/spreadsheets/d/1sck_ZeOfa0Ey-ZAFQYrmXEWReERMBFwadOm_BlaYi4s/edit#gid=1909446467)

**Comment by openstaxalina** [comment]
@lwtchu could someone on your team please review this and let me know if we have any follow-up questions? Level Access noted the severity as Critical, but note that only 8 pages are affected. It would be good for someone to review this and form a plan for resolution within the first half of September if possible.

**Comment by openstaxalina** [comment]
(noting that I edited my comment above regarding number of pages affected)

**Comment by lwtchu** [comment]
@openstaxalina Judging by the examples in the sheet, this is by design in the content. 
![Screenshot 2023-08-24 at 4 47 46 PM](https://github.com/openstax/cnx-recipes/assets/11299125/c9362ea0-cade-4690-8efa-d49d7e2e3d24)

**Comment by openstaxalina** [comment]
Copy of comms w/ Level Access:

**Query**
We reviewed the task linked above. The tables that were flagged do not have headers by design. We refer to these as "stepwise tables" - content was placed into tables to align the description of each step with the corresponding mathematical equation. For the tables with images, the table summary includes a step-by-step description.

We also have another type of stepwise table that doesn't have images -- see attached screenshot which is from [this page](https://openstax.org/books/elementary-algebra-2e/pages/10-1-solve-quadratic-equations-using-the-square-root-property).

![Screenshot 2023-08-29 at 3 44 19 PM](https://github.com/openstax/cnx-recipes/assets/7863733/c1478634-cfe8-40dc-853b-9b579ad3d9ae)

**Response - with bolding added by me**
The stepwise tables that you described, we called them “Layout tables” as they are primarily used to align the content visually and not necessarily associate the data cell to a table header. As such, this “visual” information should be conveyed to non-visual users (who are primarily the screen reader users) and **this can be done by overriding the table element roles. By providing a role=”none” or role=”presentation” attribute on the table element, you are able to maintain the visual alignment of the content while not providing any table role to the screen reader users**.

In the case of tables created using MathML, if this content is also detected during the automated scan, consider providing the role attribute to the relevant elements. Otherwise, I recommend leaving it as-is as the MathML content provides an additional way for screen reader users to navigate the content.

Hope that helped and let us know if you have any other questions.

**Comment by lwtchu** [comment]
@openstaxalina role attribute isn't valid in CNXML -- should probably be allowed. I am hesitant to apply it on a recipe level in case one of the tables deviate with respect to content (unnumbered unstyled tables are allowed to have theads)

**Comment by openstaxalina** [comment]
@lwtchu got it - so basically we will need CMs (and vendors, on titles being developed) to add either `role=presentation` or something like `class="no-header"` (and then the recipe would change that to `role=presentation`) on tables that are used for visual display/alignment only. 

I'm thinking this is a good thing to discuss at the next CE Design Systems checkin on 9/29...flagging the issue for @kswanson33, @therealmarv, and @philschatz if y'all have time to review before then! Otherwise we can just cover it next week.

**Comment by lwtchu** [comment]
@openstaxalina Right - but we'll need to allow for the `role` attribute to not error out with the validation.

**Comment by openstaxalina** [comment]
Options so far: 

1. CMs or vendors to add `role="presentation"` on `<table>` without headers; POET validation needs to be changed to allow this
2. CMs or vendors to add `class="no-header` on `<table>` without headers; the recipe would change this to `role="presentation"`
3. Somewhere in the editing (e.g. during POET push) or baking process, if <table> doesn't have <thead>, automatically add the `role="presentation"` element. We might need a way to override this, but the default would be to add `role="presentation"`

We would like to consider option 3, as this avoids the manual work of adding the class on all existing content, as well as leaves less room for error/forgetting to add to future content. I will create a card on the CE board to get a list of all `<table>` without `<thead>` so CMs/Larissa can evaluate those and determine if they should all automatically have `role="presentation"` applied.

**Comment by openstaxalina** [comment]
per discussion with @lwtchu - first we will ask CMs to make the tables with "bold" headers into true theads. Then figure out what to do with the rest that don't have actual header content.
